ckanext/datajson/plugin.py

```diff
@@ -161,31 +162,23 @@ def validator(self):
 
         return render('datajsonvalidator.html')
 
-    def generate_pdl(self):
-        # DWC this is a hack, as I couldn't get to the request parameters. For whatever reason, the multidict was always empty
-        match = re.match(r"/organization/([-a-z0-9]+)/data.json", request.path)
-        if match:
-            # set content type (charset required or pylons throws an error)
-            response.content_type = 'application/json; charset=UTF-8'
-
-            # allow caching of response (e.g. by Apache)
-            del response.headers["Cache-Control"]
-            del response.headers["Pragma"]
-            return make_pdl(match.group(1))
-        return "Invalid organization id"
-
-    def generate_edi(self):
-        # DWC this is a hack, as I couldn't get to the request parameters. For whatever reason, the multidict was always empty
-        match = re.match(r"/organization/([-a-z0-9]+)/edi.json", request.path)
```

Comment:
Same here, why do we remove the regular expression?

Reply:
Regular expression is not the best way of getting the organization id. The org id is being passed from the route anyway.